terraform/environments/example/ec2.tf

```diff
@@ -1,58 +1,40 @@
-  )
-}
-resource "aws_security_group_rule" "ingress_traffic" {
-  for_each          = local.application_data.example_ec2_sg_rules
-  description       = format("Traffic for %s %d", each.value.protocol, each.value.from_port)
-  from_port         = each.value.from_port
-  protocol          = each.value.protocol
-  security_group_id = aws_security_group.example_ec2_sg.id
-  to_port           = each.value.to_port
-  type              = "ingress"
-  cidr_blocks       = [data.aws_vpc.shared.cidr_block]
-}
-
-resource "aws_security_group_rule" "egress_traffic" {
-  for_each                 = local.application_data.example_ec2_sg_rules
-  description              = format("Outbound traffic for %s %d", each.value.protocol, each.value.from_port)
-  from_port                = each.value.from_port
-  protocol                 = each.value.protocol
-  security_group_id        = aws_security_group.example_ec2_sg.id
-  to_port                  = each.value.to_port
-  type                     = "egress"
-  source_security_group_id = aws_security_group.example_ec2_sg.id
-}
+# The security group code is in the original-ec2.tf if is required. It will be needed if there is no default one in place or if you want to add a specific one.
+# Copy the code from the above location and give it a suitable name.  
 
+# This code builds a simple EC2 instance and, if needed, increases the size of root volume (commented out at present)
 
+resource "aws_instance" "app_server" {
+  #ami           = "ami-084e8c05825742534"    # Amazon linux 64 bit
+  ami = "ami-07c2ae35d31367b3e" # Canonical, Ubuntu, 22.04 LTS, amd64
+  #ami           = "ami-0e322684a5a0074ce"   # Microsoft Windows Server 2022 Full Locale English
+  instance_type = "t3.micro"
 
-#  Build EC2 "example-ec2" 
-resource "aws_instance" "develop" {
-  # Specify the instance type and ami to be used (this is the Amazon free tier option)
-  instance_type          = local.application_data.accounts[local.environment].instance_type
-  ami                    = local.application_data.accounts[local.environment].ami_image_id
-  vpc_security_group_ids = [aws_security_group.example_ec2_sg.id]
-  subnet_id              = data.aws_subnet.private_subnets_a.id
-  monitoring             = true
-  ebs_optimized          = true
+  # The code to increase the volume size of the root volume is also in original-ec2.tf if that is needed. Add it here if required.
 
-  metadata_options {
-    http_endpoint = "enabled"
-    http_tokens   = "required"
+  tags = {
+    Name = "example-EC2"
   }
-  # Increase the volume size of the root volume
-  root_block_device {
-    volume_type = "gp3"
-    volume_size = 20
-    encrypted   = true
-  }
-  tags = merge(local.tags,
-    { Name = lower(format("ec2-%s-%s-example", local.application_name, local.environment)) }
+}
+
+# This is needed to add a role which can be used to allow access to the instance created.
+
+resource "aws_iam_role" "this" {
```

Comment:
What is using this role?  You don't need to add roles to be able to access EC2s, that is done with the user roles.